src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -337,6 +337,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <EmccArgs>&quot;$(LlvmObject)&quot; -c -o &quot;$(NativeObject)&quot; -s DISABLE_EXCEPTION_CATCHING=0</EmccArgs>
       <EmccArgs Condition="'$(NativeDebugSymbols)' != 'true'">$(EmccArgs) -O2</EmccArgs>
       <EmccArgs Condition="'$(NativeDebugSymbols)' == 'true'">$(EmccArgs) -g3</EmccArgs>
+      <EmccArgs Condition="'$(WasmHtmlTemplate)' != ''">$(EmccArgs) --shell-file $(WasmHtmlTemplate)</EmccArgs>    
```

Comment:
`--shell-file` is a link-time option, which need to be specified here:
https://github.com/dotnet/runtimelab/blob/3256ce868921db2e1bf09d0fb5aa2a86f0bc2a5e/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets#L416-L422